src/common/eeprom.cpp

```diff
@@ -277,6 +353,7 @@ void eeprom_defaults(void) {
 }
 
 variant8_t eeprom_get_var(uint8_t id) {
+    eeprom_init(); // if eeprom is not initialized, initialize it, otherwise do nothing
```

Comment:
eeprom init will happen only first time it is called
it behaves like Meyers singleton (but it not written that way because of C-like shit code in eeprom)